services/order.py

```diff
@@ -0,0 +1,54 @@
+from datetime import datetime
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.db import transaction
+from django.http import Http404
+from django.utils.timezone import make_aware
+
+from db.models import User
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, MovieSession
+
+
+def create_order(tickets: list[dict], username: str, date: str = None) -> Order:
+    try:
+        with transaction.atomic():
+            user = User.objects.get(username=username)
```

Comment:
Before creating an order, ensure that the username is unique. This is crucial to prevent data integrity issues.